### DISCUSSION TITLE
How to remove the default patterns on EntityRuler?

### DISCUSSION THREAD

**Discussion by jmaister** [explosion/spaCy#5025] [Help: Coding & Implementations] [answered]
I am working in a Jupyter Notebook for running EntityRuler:

	patterns = [
	    {"label": "LABEL_ONE", "pattern": [...]},
	    {"label": "LABEL_TWO", "pattern": [...]}
	]

	nlp = spacy.load('en')
	ruler = EntityRuler(nlp, validate=True)
	ruler.add_patterns(patterns)

	nlp.add_pipe(ruler)

	print(ruler.labels) # prints ('DRAW_MORE', 'DRAW_ONE')
	print(ruler.patterns) # prints my patterns

        t = """ my text """
        doc = nlp(t)
        labels = [ent.label_ for ent in doc.ents]

When I run the rule and get "doc.ents" it shows other labels like 'CARDINAL', 'ORDINAL', 'GPE', 'ORG', 'PERSON', etc...

How can I remove those rules from EntityRuler? I only want my rules to run on the text.

## Info about spaCy

* **spaCy version:** 2.2.3
* **Platform:** Linux-4.15.0-76-generic-x86_64-with-debian-buster-sid
* **Python version:** 3.7.4
* **Models:** en

**Comment by svlandeg** [comment, marked as answer]
The labels 'CARDINAL' etc. you're seeing, are not coming from default rules in the EntityRuler, but are in fact produced by a pretrained [NER](https://spacy.io/usage/linguistic-features#named-entities) component that is part of the pretrained [model](https://spacy.io/models/en) you're loading. To [remove](https://spacy.io/usage/processing-pipelines/#disabling) this component, you can either run your code in a `with nlp.disable_pipes("ner"):` block to disable it temporarily, or you can remove it completely by running `nlp.remove_pipe("ner")`.

The other option you have is to start with a blank model, e.g. `nlp = English()` which doesn't have any pretrained components (also no [sentence splitter](https://spacy.io/api/sentencizer)!).

**Comment by jmaister** [comment]
Thank you @svlandeg . It's working as I expected now.
